### PR TITLE
Phase2-hgx250 Comparison of results from flat file and from insitu checks

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Comparison of results from flat file and from insitu checks

#### PR validation:

Use runTheMatrix.py to test the changes

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/interface/HGCalDDDConstants.h [hunk 1]
```diff
@@ -126,6 +128,33 @@ class HGCalDDDConstants {
   std::pair<double, double> waferParameters(bool reco) const;
   std::pair<double, double> waferPosition(int wafer, bool reco) const;
   std::pair<double, double> waferPosition(int lay, int waferU, int waferV, bool reco, bool debug = false) const;
+  unsigned int waferFileSize() const { return hgpar_->waferInfoMap_.size(); }
+  int waferFileIndex(unsigned int kk) const {
+    if (kk < hgpar_->waferInfoMap_.size()) {
+      auto itr = hgpar_->waferInfoMap_.begin();
+      for (unsigned int k = 0; k < kk; ++k)
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/interface/HGCalDDDConstants.h:135]: I think you can replace this with: `std::advance(itr,kk);`

File: Geometry/HGCalCommonData/interface/HGCalDDDConstants.h [hunk 2]
```diff
@@ -126,6 +128,33 @@ class HGCalDDDConstants {
...
+        ++itr;
+      return itr->first;
+    } else
+      return 0;
+  }
+  std::tuple<int, int, int> waferFileInfo(unsigned int kk) const {
+    if (kk < hgpar_->waferInfoMap_.size()) {
+      auto itr = hgpar_->waferInfoMap_.begin();
+      for (unsigned int k = 0; k < kk; ++k)
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/interface/HGCalDDDConstants.h:144]: also here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 3]
```diff
@@ -531,9 +520,27 @@ bool HGCalDDDConstants::isValidHex8(int layer, int modU, int modV, int cellU, in
...
   }
+  return true;
+}
+
+bool HGCalDDDConstants::isValidHex8(int layer, int modU, int modV, int cellU, int cellV, bool fullAndPart) const {
+  // First check validity for a layer|wafer| of post TDR version
+  if (!isValidHex8(layer, modU, modV, fullAndPart))
+    return false;
+  int indx = HGCalWaferIndex::waferIndex(layer, modU, modV);
+  auto itr = hgpar_->typesInLayers_.find(indx);
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:531]: do you need to check for the case where indx is not found?

Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:531]: indx is basically a packed id. It is a calculated number.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@silviodonato These conflicts are due to reverting the PR bsunanda:Phase2-hgx247. I think that PR should not be reverted and then this PR can be merged

**Comment by silviodonato** [comment]
I'm sorry @bsunanda. If it can be useful in some way, I prepared a "revert of the revert" here https://github.com/cms-sw/cmssw/pull/30467

**Comment by bsunanda** [comment]
@silviodonato When #30323 will be restored?

**Comment by silviodonato** [comment]
I will restore  #30323  by merging 
https://github.com/cms-sw/cmssw/pull/30473
together with #30470 and cms-data/L1Trigger-L1THGCal#19 .
Unfortunately, the tests are failing at the moment.

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1

**Pull request merged**